dash-renderer/src/observers/prioritizedCallbacks.ts

```diff
@@ -0,0 +1,104 @@
+import {
+    flatten,
+    map,
+    reduce,
+    uniq,
+    pluck,
+    partition,
+    includes
+} from 'ramda';
+
+import { IStoreState } from "../store";
+
+import {
+    aggregateCallbacks,
+    removePrioritizedCallbacks,
+    addExecutingCallbacks,
+    executeCallback
+} from '../actions/callbacks';
+
+import { stringifyId } from '../actions/dependencies';
+
+import {
+    combineIdAndProp
+} from '../actions/dependencies_ts';
+
+import isAppReady from '../actions/isAppReady';
+
+import {
+    ICallback,
+    IExecutingCallback,
+    ICallbackProperty
+} from '../types/callbacks';
+import { IStoreObserverDefinition } from '../StoreObserver';
+
+const observer: IStoreObserverDefinition<IStoreState> = {
+    observer: async ({
+        dispatch,
+        getState
+    }) => {
+        const { callbacks: { executing, watched }, config, hooks, layout, paths } = getState();
+        let { callbacks: { prioritized } } = getState();
+
+        const available = Math.max(
+            0,
+            6 - executing.length - watched.length
+        );
```

Comment:
callbacks that `can` be executed are promoted from `requested` to `prioritized` - from `prioritized` callbacks will be executed - conditions can be applied to limit the number of concurrent callbacks or bundle them, etc.